src/main/main.ts

```diff
@@ -253,20 +258,31 @@ async function initializeApp() {
     webEnable(win.webContents);
     createTray(path.join(app.getAppPath(), logoImage));
 
-    const u = await checkForUpdates(standalone);
-    if (u && !isV2) update(u, updateOptions);
-    else if (u && isV2)
-      ipcMain.handle("start update", async () => {
-        await update(u, updateOptions);
-      });
+    const context = await checkUpdateRequired(
+      standalone,
+      process.platform,
+      netenv,
+      getConfig("PeerStrings"),
+      baseURL,
+      getConfig("AppProtocolVersion"),
+      getConfig("TrustedAppProtocolVersionSigners")
```

Comment:
Is there a reason we need to keep these arguments? it seems like they can be initiated from `check.ts`.

Reply:
umm... I did think "Should make Config object contain default values later" in this work
but now init some configs in `check.ts` is no problem

Reply:
Done